src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.cs

```diff
@@ -163,23 +216,55 @@ public async Task<ImmutableArray<Diagnostic>> GetDocumentDiagnosticsAsync(Docume
             }
 
             var getDiagnosticsTask = State.DiagnosticProvider.GetDocumentDiagnosticsAsync(document, this.CancellationToken);
-            return await GetFilteredDiagnosticsAsync(getDiagnosticsTask, this.DiagnosticIds).ConfigureAwait(false);
+            return await GetFilteredDiagnosticsAsync(getDiagnosticsTask, this.DiagnosticIds, filterSpan: null).ConfigureAwait(false);
         }
 
-        private static async Task<ImmutableArray<Diagnostic>> GetFilteredDiagnosticsAsync(Task<IEnumerable<Diagnostic>> getDiagnosticsTask, ImmutableHashSet<string> diagnosticIds)
+        private static async Task<ImmutableArray<Diagnostic>> GetFilteredDiagnosticsAsync(
+            Task<IEnumerable<Diagnostic>> getDiagnosticsTask,
+            ImmutableHashSet<string> diagnosticIds,
+            TextSpan? filterSpan)
         {
             if (getDiagnosticsTask != null)
             {
                 var diagnostics = await getDiagnosticsTask.ConfigureAwait(false);
                 if (diagnostics != null)
                 {
-                    return diagnostics.Where(d => d != null && diagnosticIds.Contains(d.Id)).ToImmutableArray();
+                    return diagnostics.Where(d => d != null && diagnosticIds.Contains(d.Id)
+                        && (filterSpan == null || filterSpan.Value.Contains(d.Location.SourceSpan))).ToImmutableArray();
                 }
             }
 
             return ImmutableArray<Diagnostic>.Empty;
         }
 
+        /// <summary>
+        /// Gets all the diagnostics in the given <paramref name="filterSpan"/> for the given <paramref name="document"/> filtered by <see cref="DiagnosticIds"/>.
+        /// </summary>
+        public async Task<ImmutableArray<Diagnostic>> GetDocumentSpanDiagnosticsAsync(Document document, TextSpan filterSpan)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+
+            if (this.Project.Language != document.Project.Language)
+            {
+                return ImmutableArray<Diagnostic>.Empty;
+            }
+
+            Task<IEnumerable<Diagnostic>>? getDiagnosticsTask;
+            if (State.DiagnosticProvider is FixAllContext.SpanBasedDiagnosticProvider spanBasedDiagnosticProvider)
+            {
+                getDiagnosticsTask = spanBasedDiagnosticProvider.GetDocumentSpanDiagnosticsAsync(document, filterSpan, this.CancellationToken);
+            }
+            else
+            {
+                getDiagnosticsTask = State.DiagnosticProvider.GetDocumentDiagnosticsAsync(document, this.CancellationToken);
```

Comment:
is there somethign that post-filters the diagnostics from this to the passed in filterspan?  i'm not seeing it.